### ISSUE TITLE
[Python] NamedTuple SolutionOutputs is not actually a NamedTuple

### ISSUE THREAD

**Issue by Interpause** [google/mediapipe#3257]
**System information** (Please provide as much relevant information as possible)

- Have I written custom code (as opposed to using a stock example script provided in MediaPipe): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04, Android 11, iOS 14.4): NA
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: NA
- Browser and version (e.g. Google Chrome, Safari) if the issue happens on browser: NA
- Programming Language and version ( e.g. C++, Python, Java): Python
- [MediaPipe version](https://github.com/google/mediapipe/releases): 0.8.9.1
- Bazel version (if compiling from source): NA
- Solution ( e.g. FaceMesh, Pose, Holistic ): Pose
- Android Studio, NDK, SDK versions (if issue is related to building in Android environment): NA
- Xcode & Tulsi version (if issue is related to building for iOS): NA

**Describe the current behavior:**

When calling `_asdict()` on `SolutionOutputs` (given it is a `NamedTuple`), it says `TypeError: _asdict() missing 1 required positional argument: 'self'`. `SolutionOutputs` obtained by running `pose.process(image)`.

**Describe the expected behavior:**

Calling `_asdict()` should successfully convert the results object into a dictionary.

**Standalone code to reproduce the issue:**
Provide a reproducible test case that is the bare minimum necessary to replicate the problem. If possible, please share a link to Colab/repo link /any notebook:

```py
import cv2
import mediapipe as mp
mp_pose = mp.solutions.pose

with mp_pose.Pose(
    static_image_mode=True,
    model_complexity=2,
    enable_segmentation=True,
    min_detection_confidence=0.5) as pose:
    image = cv2.imread('test.png')
    image_height, image_width, _ = image.shape
    # Convert the BGR image to RGB before processing.
    results = pose.process(cv2.cvtColor(image, cv2.COLOR_BGR2RGB))

    if not results.pose_landmarks:
      continue
    print(results._asdict())
    
```

**Other info / Complete Logs :**
Looking into the source code (`mediapipe/python/solution_base.py`), I noticed that a new `NamedTuple` factory is being created each time during `process()`, and then the results are added to the newly created classs via `setattr()`. This isn't the correct way to use a NamedTuple. The correct way is:

```py
solution_outputs = collection.namedtuple('SolutionOutputs', keys)

results = solution_outputs(pose_landmarks=..., ...)
```

**Comment by sureshdagooglecom** [comment]
Hi @Interpause ,
Could you please elaborate w.r.t above issue to investigate further.

**Comment by Hjosmelvy** [comment]
I created a possible fix in this pull request #4024

**Comment by kuaashish** [comment]
Hello @Interpause,
We are upgrading the [MediaPipe Legacy Solutions](https://developers.google.com/mediapipe/solutions/guide#legacy) to [new MediaPipe solutions](https://developers.google.com/mediapipe/solutions) However, the libraries, documentation, and source code for all the MediapPipe Legacy Solutions will continue to be available in our [GitHub repository](https://github.com/google/mediapipe) and through library distribution services, such as [Maven](https://maven.google.com/web/index.html?#com.google.mediapipe) and [NPM](https://www.npmjs.com/search?q=%40mediapipe).

You can continue to use those legacy solutions in your applications if you choose. Though, we would request you to check [new MediaPipe solutions](https://developers.google.com/mediapipe/solutions) which can help you more easily build and customize ML solutions for your applications. These new solutions will provide a superset of capabilities available in the legacy solutions. Thank you
